src/lib/message/L1ToL2Message.ts

```diff
@@ -245,17 +246,50 @@ export class L1ToL2MessageReader extends L1ToL2Message {
     const creationReceipt =
       retryableCreationReceipt ||
       (await this.l2Provider.getTransactionReceipt(this.retryableCreationId))
-
-    if (creationReceipt) {
-      const l2Receipt = new L2TransactionReceipt(creationReceipt)
-      const redeemEvents = l2Receipt.getRedeemScheduledEvents()
+    if (creationReceipt){
+      const creationBlockNumber = creationReceipt.blockNumber
```

Comment:
I'm not sure what the advantage is of replacing the old code with this new section. Both only fetch the first redeem attempt, but the previous version did it without any log calls at all.

I think what we need is a new function to fetch the first successful redeem, instead of just the first scheduled one. For this you need to find the last redeem that took place, and check if it was successful.

Reply:
There are no redeem attempt in nitro if it the auto redeem fail.

Reply:
A user can later call `redeem` manually on ArbRetryable.sol, which will schedule a new redeem.

Reply:
The old code look for the redeem attempt only in the ticket creation tx, so it won't be able to find manual redemption.

Reply:
Yeah, but I think the new code and old are trying to do different jobs, the old code returns the initial redeem attempt. The new code should be looking for the first successful redeem. So I think they should be separated into different functions. 

I think the new code doesn't quite get the first successful redeem, as it misses successful redeems that happen outside the current block interval being checked. The code does the following:

1. Looks for any redeem attempts in the first 1000 blocks
2. If it finds one (it definitely will since the ticket creation itself causes one redeem attempt), it breaks from the while loop
3. It then returns (but only if there's one redeem attempt: https://github.com/OffchainLabs/arbitrum-sdk/pull/17/files#diff-44f6a12b564108aa7000a7e940d0918990f51cd4bd32feacff3a57d8448728a6R287)

So I think scenarios like the following dont get found:
block 0: ticket is created, redeem scheduled but failed 
block 500: ticket redeemed manually, but redemption failed
block 1001: ticket redeemed manualy, redemption successful.

I think the new code would return the redemptions from block 0 and block 500, then break from the while loop and miss the successful redemption on block 1001

Reply:
> If it finds one (it definitely will since the ticket creation itself causes one redeem attempt), it breaks from the while loop
This is incorrect, at least if the redeem attempt failed due to low gas price there will not be redeem attempt.

Reply:
Yep, good point

Reply:
There are 3 case upon retryable creation
1. Auto redeem success
2. Auto redeem failed due to gas parameters
3. Auto redeem failed due to l2 call revert
1 and 3 will emit `RedeemScheduled` that we can easily catch
2 will not emit `RedeemScheduled` and will require a log scan, which this PR solves

We will also need to add a function that return the first successful attempt, correct?